From 858837a9f8b9c1586aae29bf26551aa2f7e900ef Mon Sep 17 00:00:00 2001
From: AnilKumar Ch <anilkumar@ti.com>
Date: Thu, 7 Jun 2012 17:11:27 +0530
Subject: [PATCH 1011/1092] ARM: OMAP2+: am335x: platform data addition to add
 voltage ramp delay

This patch adds the minimum delay required to ramp the MPU voltage
from minimum OPP to maximum OPP, this is required only for BeagleBone.

MPU voltage transition from OPP50 to OPPTURBO took minimum of 70ms
to ramp completely when observed using an oscilloscope.

The reason for extended ramp time requirement on BeagleBone is not known.
This ramp time varies from board - board, if the board hangs with this
70msec delay also then try to increase this value.

Signed-off-by: AnilKumar Ch <anilkumar@ti.com>
---
 arch/arm/mach-omap2/board-am335xevm.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/arch/arm/mach-omap2/board-am335xevm.c b/arch/arm/mach-omap2/board-am335xevm.c
index 444061c..2156d00 100644
--- a/arch/arm/mach-omap2/board-am335xevm.c
+++ b/arch/arm/mach-omap2/board-am335xevm.c
@@ -1284,6 +1284,20 @@ static struct regulator_consumer_supply tps65217_ldo4_consumers[] = {
 	},
 };
 
+/*
+ * FIXME: Some BeagleBones reuire a ramp_delay to settle down the set
+ * voltage from 0.95v to 1.25v. By default a minimum of 70msec is set
+ * based on experimentation. This will be removed/modified to exact
+ * value, once the root cause is known.
+ *
+ * The reason for extended ramp time requirement on BeagleBone is not
+ * known and the delay varies from board - board, if the board hangs
+ * with this 70msec delay then try to increase the value.
+ */
+static struct tps65217_rdelay dcdc2_ramp_delay = {
+	.ramp_delay = 70000,
+};
+
 static struct regulator_init_data tps65217_regulator_data[] = {
 	/* dcdc1 */
 	{
@@ -1309,6 +1323,7 @@ static struct regulator_init_data tps65217_regulator_data[] = {
 		},
 		.num_consumer_supplies = ARRAY_SIZE(tps65217_dcdc2_consumers),
 		.consumer_supplies = tps65217_dcdc2_consumers,
+		.driver_data = &dcdc2_ramp_delay,
 	},
 
 	/* dcdc3 */
-- 
1.7.11.2

